vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc

```diff
@@ -132,46 +134,34 @@ bool Twitter::IsExcludedPath(const std::string& path) {
     return true;
   }
 
-  const std::vector<std::string> paths({
-      "/",
-      "/settings",
-      "/explore",
-      "/notifications",
-      "/messages",
-      "/logout",
-      "/search",
-      "/about",
-      "/tos",
-      "/privacy",
-      "/home"
+  const std::vector<std::string> screen_names({
+      "settings",
+      "explore",
+      "notifications",
+      "messages",
+      "logout",
+      "search",
+      "about",
+      "tos",
+      "privacy",
+      "home",
     });
 
-  for (std::string str_path : paths) {
-    if (str_path == path || str_path + "/" == path) {
-      return true;
-    }
+  GURL url(TWITTER_BASE_URL + path);
+  if (!url.is_valid()) {
+    return true;
   }
 
-  const std::vector<std::string> patterns({
-    "/i/",
-    "/account/",
-    "/compose/",
-    "/?login",
-    "/?logout",
-    "/who_to_follow/",
-    "/hashtag/",
-    "/settings/"
-  });
-
-  for (std::string str_path : patterns) {
-    if (base::StartsWith(path,
-                         str_path,
-                         base::CompareCase::INSENSITIVE_ASCII)) {
-      return true;
+  for (net::QueryIterator it(url); !it.IsAtEnd(); it.Advance()) {
+    if (it.GetKey() == "screen_name") {
```

Comment:
why do we first check for screen_name in the  url and only then we process paths?

Reply:
will https://twitter.com/explore be skipped in this case?

Reply:
The C++ backend will never see a standard URL like `https://twitter.com/emerick` now. The Twitter content script translates all such URLs to `https://twitter.com/intent/user?user_id=${userId}&screen_name=${screenName}`. So we're no longer looking at the URL path when trying to determine whether or not to exclude it, but instead we're looking at the `screen_name` query-string parameter.

Reply:
And yes, `explore` will be excluded in this case.

Reply:
so example of url is https://twitter.com/notifications?user_id=${userId}&screen_name=${screenName} as well?

Reply:
If a user visits `https://twitter.com/notifications` the content script will redirect them to `https://twitter.com/intent/user?user_id=undefined&screen_name=notifications`, so the C++ backend will only see the latter URL. In this specific case, the `user_id` is `undefined` because the Twitter API doesn't have an associated user ID for any of the "special" Twitter handles like `notifications` and `explore`.

If we don't like that, we could perform the exclusion test in the content script and then never even bother sending "special" URLs to the C++ backend. I didn't want to get too invasive with the changes here though, given that architectural changes are already planned anyway.

Reply:
all good